spec/lib/rex/proto/kerberos/model/krb_cred_spec.rb

```diff
@@ -0,0 +1,132 @@
+    "\x94\xcf\x88\x27\x34\xdb\x3a\xdf\xb0\xf3\x3a\xb6\xec\x73\x51\xd8" \
+    "\xa5\xfe\x42\xb3\xce\x54\x77\xec\x14\xeb\x90\x77\xe7\x9c\x22\xbe" \
+    "\x59\xe0\x2c\xd8\xf1\x14\xf8\x0a\x7a\xd3\xe5\xf3\x81\x78\xf4\x7a" \
+    "\x1e\xe1\xda\xb8\xb0\xf9\xbe\xc5\x56\x21\x2d\x6e\x19\x8b\xb3\xbe" \
+    "\xd8\x1b\x7b\x75\x07\xa2\x66\x61\x2f\x59\x32\x91\xa8\x7b\xe2\x20" \
+    "\xbc\xa5\x27\xd3\xd8\x80\x90\x6f\x02\xf3\xcc\x8f\xdd\x70\xdd\x38" \
+    "\x1f\x70\xb6\xfa\x56\x05\xbc\x8b\x30\xc4\x73\xf0\x6c\xd4\x5f\x66" \
+    "\xe4\xa8\xfb\xda\x3f\xef\x9f\x9e\xba\xbc\x22\xb3\x32\x42\xc2\x4f" \
+    "\x88\xf7\x20\xf7\x04\x82\x38\xd1\xc0\x8f\xef\xc7\xa0\xd1\xdb\xdf" \
+    "\x06\x5a\xe0\x2a\xe5\xe6\x87\x87\x97\x65\xe0\x3b\x02\xe0\x9b\xd6" \
+    "\x05\x99\xe8\x2e\xef\xad\x0c\x99\xc5\x53\x09\xef\x8e\x8f\x4c\x54" \
+    "\xe2\xac\xc2\x66\xfb\x2e\x22\x2c\xe7\x37\xf1\xb7\x57\x2a\x61\x19" \
+    "\xba\xa2\xc6\x67\x8c\xc5\xb8\x1a\x5e\xed\xa3\x81\xd7\x30\x81\xd4" \
+    "\xa0\x03\x02\x01\x00\xa2\x81\xcc\x04\x81\xc9\x7d\x81\xc6\x30\x81" \
+    "\xc3\xa0\x81\xc0\x30\x81\xbd\x30\x81\xba\xa0\x1b\x30\x19\xa0\x03" \
+    "\x02\x01\x17\xa1\x12\x04\x10\x70\x74\x58\x44\x6c\x73\x44\x78\x4d" \
+    "\x4e\x6a\x48\x71\x65\x4a\x70\xa1\x0a\x1b\x08\x44\x57\x2e\x4c\x4f" \
+    "\x43\x41\x4c\xa2\x17\x30\x15\xa0\x03\x02\x01\x01\xa1\x0e\x30\x0c" \
+    "\x1b\x0a\x66\x61\x6b\x65\x5f\x6d\x79\x73\x71\x6c\xa3\x07\x03\x05" \
+    "\x01\xa1\x40\x00\x00\xa5\x11\x18\x0f\x32\x30\x32\x32\x30\x38\x30" \
+    "\x38\x31\x33\x30\x32\x35\x38\x5a\xa6\x11\x18\x0f\x32\x30\x33\x32" \
+    "\x30\x38\x30\x35\x31\x33\x30\x32\x35\x38\x5a\xa7\x11\x18\x0f\x32" \
+    "\x30\x33\x32\x30\x38\x30\x35\x31\x33\x30\x32\x35\x38\x5a\xa8\x0a" \
+    "\x1b\x08\x44\x57\x2e\x4c\x4f\x43\x41\x4c\xa9\x28\x30\x26\xa0\x03" \
+    "\x02\x01\x01\xa1\x1f\x30\x1d\x1b\x08\x4d\x53\x53\x71\x6c\x53\x76" \
+    "\x63\x1b\x11\x64\x63\x31\x2e\x64\x77\x2e\x6c\x6f\x63\x61\x6c\x3a" \
+    "\x31\x34\x33\x33"
+  end
+
+  subject(:krb_cred) do
+    Rex::Proto::Kerberos::Model::KrbCred.decode(sample_kirbi)
+  end
+
+  describe '#decode' do
+    it { is_expected.to be_a(Rex::Proto::Kerberos::Model::KrbCred) }
+    it { expect(subject.pvno).to eq(5) }
+    it { expect(subject.msg_type).to eq(22) }
+    it { expect(subject.tickets).to be_an(Array) }
+    it { expect(subject.tickets.length).to be(1) }
+    it { expect(subject.tickets).to all be_a(Rex::Proto::Kerberos::Model::Ticket) }
+    it { expect(subject.enc_part).to be_an(Rex::Proto::Kerberos::Model::EncryptedData) }
+    it "successfully decodes the EncKrbCredPart" do
+      enc_krb_part = Rex::Proto::Kerberos::Model::EncKrbCredPart.decode(subject.enc_part.cipher)
+      expect(enc_krb_part).to be_an(Rex::Proto::Kerberos::Model::EncKrbCredPart)
+    end
+  end
+
+  describe '#save_credential_to_file' do
+    let(:kirbi_file) { Tempfile.new('test_kirbi_file_save.kirbi') }
+
+    it { expect { subject.save_credential_to_file(kirbi_file.path) }.not_to raise_error }
```

Comment:
should we verify that the kirbi file is at least not empty here? 👀

Reply:
I mean honestly looking at the function all this is testing is that I know how to use the `File` api to write to disk, pretty worthless test I'd be more tempted to just delete it